Data-Structures-Trie.md

```diff
@@ -0,0 +1,99 @@
+# Trie
+
+## Introduction to Trie
+
+The word trie is an inflix of the word "re**trie**val", because the trie can find a single word in a dictionary with only a prefix of the word.
+Trie is an efficient data retrieval data structure, using trie, search complexities can be brought to an optimal limit, i.e. length of the string.
+It is a multi-way tree structure useful for storing strings over an alphabet.
+It has been used to store large dictionaries (do not confuse this with the dictionaries of python) of any language, say English, words in spell-checking programs.
+However, the penalty on tries is the storage requirement.
+
+## What is a trie?
+
+A trie is a tree like data structure which stores strings, and helps you find the data associated with that string using the prefix of the string. 
```

Comment:
@alayek I have added one in the implementation section, should I move it up here?